flask_multipass_cern.py

```diff
@@ -98,10 +112,21 @@ def get_members(self):
     def has_member(self, identifier):
         cache_key = f'flask-multipass-cern:{self.provider.name}:groups:{identifier}'
         all_groups = self.provider.cache and self.provider.cache.get(cache_key)
-        if all_groups is None:
-            all_groups = {g.name.lower() for g in self.provider.get_identity_groups(identifier)}
-            if self.provider.cache:
-                self.provider.cache.set(cache_key, all_groups, 1800)
+        should_refresh = self.provider.cache and self.provider.cache.get(f'{cache_key}:timestamp') is None
+
+        if all_groups is None or should_refresh:
+            try:
+                all_groups = {g.name.lower() for g in self.provider.get_identity_groups(identifier)}
+                if self.provider.cache:
+                    self.provider.cache.set(cache_key, all_groups, timeout=CACHE_LONG_TTL)
+                    self.provider.cache.set(f'{cache_key}:timestamp', datetime.now(), timeout=CACHE_TTL)
+            except HTTPError as error:
+                logger = logging.getLogger(self.provider.settings['logger_name'])
+                if all_groups is None:
+                    logger.exception(f'Get user groups failed, access will be denied: {error}')
```

Comment:
better to pass the error separately; otherwise sentry will create a new issue for each error message:

    logger.exception('Get user groups failed, access will be denied: %s', error)

But actually, as you log as `.exception()` it may not be needed at all since it already includes the current exception in the log record..